magda-authorization-api/src/test/buildCreateApiRouter.spec.ts

```diff
@@ -0,0 +1,505 @@
+
+                await jsc.assert(
+                    jsc.forall(userDataArb, async userData => {
+                        try {
+                            const req = request(app)
+                                .post("/private/users")
+                                .send(userData);
+
+                            setMockRequestSession(req, standardUserId);
+
+                            const res = await req.then(res => res);
+
+                            expect(res.status).to.equal(403);
+                            expect(mockUserDataStore.countRecord()).to.equal(
+                                currentMockUserStoreSize
+                            );
+
+                            return true;
+                        } catch (e) {
+                            throw e;
+                        }
+                    })
+                );
+            });
+
+            it("should create a new user if called as an admin user", async () => {
+                const adminUserId = mockUserDataStore.getRecordByIndex(0).id;
+
+                await jsc.assert(
+                    jsc.forall(userDataArb, async userData => {
+                        try {
+                            const req = request(app)
+                                .post("/private/users")
+                                .send(userData);
+
+                            setMockRequestSession(req, adminUserId);
+
+                            const res = await req.then(res => res);
+                            expect(res.status).to.equal(200);
+                            expect(res.body).to.be.a("object");
+                            expect(res.body.id).to.be.a("string");
+
+                            const newUserId: string = res.body.id;
+                            expect(res.body).to.deep.equal({
+                                ...userData,
+                                id: res.body.id
+                            });
+
+                            const users = mockUserDataStore.getRecordByUserId(
+                                newUserId
+                            );
+                            expect(users).to.be.an("array").that.is.not.empty;
+                            expect(users[0]).to.deep.equal({
+                                ...userData,
+                                id: newUserId
+                            });
+
+                            newlyCreatedUsers.push(users[0]);
+
+                            return true;
+                        } catch (e) {
+                            throw e;
+                        }
+                    })
+                );
+            });
+        });
+    });
+
+    describe("GET /private/users/lookup", () => {
+        silenceErrorLogs(() => {
+            it("should return 401 status code if requested without sepecifying user ID", async () => {
+                await jsc.assert(
+                    jsc.forall(
+                        jsc.elements(newlyCreatedUsers),
```

Comment:
So if this test got moved above `should create a new user if called as an admin user` this'd fail right?

I'd really strongly prefer if the unit tests didn't depend on each other. Could you use `userDataGen` (or `jsc.array(userDataGen)` when you need multiple users) to generate users for each spec and insert them into `mockUserDataStore` at the start?

Reply:
Yes. As per the other commit, I think I will re-structure the tests.